### PR TITLE
Rename 'current' type port to 'continuous'; make references to 'buffer' consistent

### PR DESCRIPTION
**Pull request by clinssen**
Pure refactoring without behavioural change.

For third-factor synaptic plasticity rules, our aim is to re-used the 'continuous time' input ports, which are now used exclusively for currents (see https://github.com/nest/nestml/issues/646). Thus, the keyword 'current' was renamed to 'continuous'.

Also, references to 'buffer' were renamed to 'input port', referring to the named input port in the model input block. (References to the actual NEST C++-level buffers were left unchanged.)

### CODE REVIEW

File: pynestml/generated/PyNestMLParserVisitor.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Generated from PyNestMLParser.g4 by ANTLR 4.9.1
+# Generated from PyNestMLParser.g4 by ANTLR 4.7.1
```
Comment (pnbabu) [inline pynestml/generated/PyNestMLParserVisitor.py:1]:
```suggestion
# Generated from PyNestMLParser.g4 by ANTLR 4.9.1
```

File: tests/invalid/CoCoConvolveNotCorrectlyParametrized.nestml [hunk 2]
```diff
@@ -6,8 +6,7 @@ CoCoConvolveNotCorrectlyParametrized.nestml
 Description
 +++++++++++
 
-This model is used to test if broken CoCos are identified correctly. Here, if convolve has been correctly
-provided with a initial_block variable and a spike buffer.
+This model is used to test if broken CoCos are identified correctly. Here, if convolve has been correctly provided with a initial_block variable and a spike input port.
```
Comment (pnbabu) [inline tests/invalid/CoCoConvolveNotCorrectlyParametrized.nestml:9]:
```suggestion
This model is used to test if broken CoCos are identified correctly. Here, if convolve has been correctly provided with a state block variable and a spike input port.
```

File: tests/invalid/CoCoConvolveNotCorrectlyProvided.nestml [hunk 3]
```diff
@@ -6,8 +6,7 @@ CoCoConvolveNotCorrectlyProvided.nestml
 Description
 +++++++++++
 
-This model is used to test if broken CoCos are identified correctly. Here, if convolve has been correctly
-provided with a initial_block variable and a spike buffer.
+This model is used to test if broken CoCos are identified correctly. Here, if convolve has been correctly provided with a initial_block variable and a spike input port.
```
Comment (pnbabu) [inline tests/invalid/CoCoConvolveNotCorrectlyProvided.nestml:9]:
```suggestion
This model is used to test if broken CoCos are identified correctly. Here, if convolve has been correctly provided with a state block variable and a spike input port.
```

File: tests/valid/CoCoConvolveNotCorrectlyParametrized.nestml [hunk 4]
```diff
@@ -6,8 +6,7 @@ CoCoConvolveNotCorrectlyParametrized.nestml
 Description
 +++++++++++
 
-This model is used to test if broken CoCos are identified correctly. Here, if convolve has been correctly
-provided with a initial_block variable and a spike buffer.
+This model is used to test if broken CoCos are identified correctly. Here, if convolve has been correctly provided with a initial_block variable and a spike input port.
```
Comment (pnbabu) [inline tests/valid/CoCoConvolveNotCorrectlyParametrized.nestml:9]:
```suggestion
This model is used to test if broken CoCos are identified correctly. Here, if convolve has been correctly provided with a state block variable and a spike input port.
```

File: tests/valid/CoCoConvolveNotCorrectlyProvided.nestml [hunk 5]
```diff
@@ -6,8 +6,7 @@ CoCoConvolveNotCorrectlyProvided.nestml
 Description
 +++++++++++
 
-This model is used to test if broken CoCos are identified correctly. Here, if convolve has been correctly
-provided with a initial_block variable and a spike buffer.
+This model is used to test if broken CoCos are identified correctly. Here, if convolve has been correctly provided with a initial_block variable and a spike input port.
```
Comment (pnbabu) [inline tests/valid/CoCoConvolveNotCorrectlyProvided.nestml:9]:
```suggestion
This model is used to test if broken CoCos are identified correctly. Here, if convolve has been correctly provided with a state block variable and a spike input port.
```

**Comment by clinssen** [comment]
Some further discussion points:
- syntax/naming (especially the name "continuous"; the time base is, after all, discrete in our simulation)
- NEST CurrentEvent handler should be generated if and only if the unit type of the input port is (convertible to) Ampère.

**Comment by clinssen** [comment]
The CI test failure appears to be due to an unrelated issue: https://github.com/nest/nest-simulator/pull/2045

File: tests/nest_tests/nest_integration_test.py [hunk 6]
```diff
@@ -272,11 +274,16 @@ def _test_model_subthreshold(self, referenceModel, testant, gsl_error_tol, toler
...
                 plt.savefig("/tmp/nestml_models_library_[" + referenceModel + "]_f-I_curve" + fname_snip + ".png")
+                plt.close(fig)
 
         print(testant + " PASSED")
 
     def _test_model(self, referenceModel, testant, gsl_error_tol, tolerance=0.000001, nest_ref_model_opts=None, custom_model_opts=None):
 
+        print("Now testing model: " + str(testant) + " (reference model: " + str(referenceModel) + ")")
+        import logging
+        logging.info("Now testing model: " + str(testant) + " (reference model: " + str(referenceModel) + ")")
```
Comment (pnbabu) [inline tests/nest_tests/nest_integration_test.py:285]: Could this be removed?

**Review by pnbabu — APPROVED** [review]
I have a minor comment. Otherwise, it looks good to me.

**Comment by clinssen** [comment]
Thanks for the check!

**Pull request merged**